addons/hr_holidays/tests/test_leave_requests.py

```diff
@@ -1140,6 +1142,16 @@ def test_default_request_date_timezone(self):
             taken into account when requesting a leave.
         """
         self.user_employee.tz = 'Asia/Hong_Kong'  # UTC +08:00
+        allocation = self.env['hr.leave.allocation'].create({
+            'name': 'Allocation for user_employee',
+            'employee_id': self.employee_emp_id,
+            'holiday_status_id': self.holidays_type_2.id,
+            'number_of_days': 5,
+            'state': 'confirm',
+            'date_from': '2024-01-01',
+            'date_to': '2024-12-31',
+        })
+        allocation.action_validate()
```

Comment:
If this is not related to leave types requiring allocation specifically, switch the leave type of the leaves to be for a leave type with no allocation required instead, same below